docs/designs/upgrade-design/upgrade-design-doc.md

```diff
@@ -0,0 +1,653 @@
+- **Risk Mitigation**: Test new API versions in dev/staging before production migration
+- **Backward Compatibility**: Multiple cluster API versions supported simultaneously
+- **Controlled Deprecation**: API versions deprecated gradually over multiple operator releases
+
+**Cross-Team Coordination**: Database Admins coordinate migration planning with Database Developers for execution, while Kubernetes Admins provide the infrastructure foundation.
+
+**API Version Examples**: See [commands.md](./commands.md) for detailed workflow commands and API version migration examples.
+
+## Component-Specific Upgrade Considerations
+
+While all components upgrade together, each has specific characteristics:
+
+#### Gateway Image Upgrade (API Version Dependent)
+- **State**: **Stateless** - Gateway containers have no persistent state
+- **Impact**: Medium (rolling restart of pods when API version changes)
+- **Risk**: Low - No data loss risk, only temporary connection disruption
+- **Risk Mitigation**: Multiple standby instances with local HA ensure zero-downtime rolling restart; Gateway and PostgreSQL containers run in same pod, sharing HA benefits
+- **Version Behavior**: Gateway features may differ between cluster API v1 and v2
+
+#### PostgreSQL Database Upgrade (API Version Dependent)  
+- **State**: **Stateful** - PostgreSQL contains persistent application data
+- **Impact**: Variable (depends on API version differences)
+- **Risk**: Variable - Data migration only required for breaking schema changes
+- **Risk Mitigation**: CNPG managed HA with supervised rolling updates for zero-downtime upgrades
+- **HA Strategy**: 3-instance clusters (1 primary + 2 standby servers) with CNPG-controlled failover sequence
+- **Categories**:
+  - **Same PostgreSQL Version**: Cluster API v1 → v2 with same PG version (low risk, configuration change only)
+  - **Minor PostgreSQL Version**: Different PG minor versions between API versions (medium risk, CNPG rolling restart with automatic switchover)
+  - **Major PostgreSQL Version**: Different PG major versions between API versions (high risk, data migration required with blue-green procedures)
+
+#### DocumentDB Postgres Extension Upgrade (API Version Dependent)
+- **State**: **Stateful** - Extension may have API-specific features
+- **Impact**: Variable (depends on extension differences between API versions)
+- **Risk**: Variable - Extension schema changes only if API versions require different features
+- **Risk Mitigation**: Extension compatibility testing during operator upgrade; rollback capability maintains previous extension versions if needed
+- **Categories**:
+  - **Compatible Extension**: Same extension version for both API versions (low risk)
+  - **Enhanced Extension**: New features added for v2 API (medium risk, backward compatible)
+  - **Breaking Extension Changes**: Schema modifications required for v2 API (high risk, requires migration planning)
```

Comment:
usually an extension manages the migration and we assume for k8s that has been carefully planned and orchestrated by the extension devs